sensors/baro/lps25xx.c

```diff
@@ -16,61 +16,151 @@
 #include <unistd.h>
 #include <sys/time.h>
 #include <sys/threads.h>
+#include <spi.h>
+#include <spi-msg.h>
 
 #include "../sensors.h"
 
+/* self identification register */
+#define REG_WHOAMI     0x0f
+#define REG_VAL_WHOAMI 0xbd
+
+/* resolution configuration register */
+#define REG_RES_CONF 0x10
+#define VAL_AVGT_8   0x00
+#define VAL_AVGT_32  0x04
+#define VAL_AVGT_128 0x08
+#define VAL_AVGT_512 0x0c
+#define VAL_AVGP_8   0x00
+#define VAL_AVGP_32  0x01
+#define VAL_AVGP_128 0x02
+#define VAL_AVGP_512 0x03
+
+/* control register 1 */
+#define REG_CTRL_REG1      0x20
+#define VAL_PD_ACTIVE_MODE 0x80
+#define VAL_ODR_ONESHOT    0x00
+#define VAL_ODR_1HZ        0x10
+#define VAL_ODR_7HZ        0x20
+#define VAL_ODR_12HZ       0x30
+#define VAL_ODR_25HZ       0x40
+
+/* data storage addresses */
+#define REG_DATA_OUT         0x28
+#define SPI_READ_BIT         0x80
+#define SPI_AUTOADDRINCR_BIT 0x40
+
+/* sensors data sizes */
+#define SENSOR_OUTPUT_SIZE 5
+
+/* conversions */
+#define SENSOR_CONV_PASCALS 0.024414062F /* conversion from sensor value to pascals; = (100 * 1/4096) */
+#define SENSOR_CONV_KELV    0.002083333F /*conversion from sensor value to kelvin */
+#define SENSOR_KELV_OFFS    315.65F      /* temperature measurement offset + kelvin to celsius offset */
 
 typedef struct {
-	sensor_event_t evBaro;
+	spimsg_ctx_t spiCtx;
+	sensor_event_t evtBaro;
 	char stack[512] __attribute__((aligned(8)));
 } lps25xx_ctx_t;
 
 
-static inline time_t lps25xx_timeMsGet(void)
+static uint32_t translatePress(uint8_t lbyte, uint8_t mbyte, uint8_t hbyte)
 {
-	time_t now;
-	gettime(&now, NULL);
+	uint32_t val = 0;
+	val |= ((uint32_t)lbyte) | ((uint32_t)mbyte << 8) | ((uint32_t)hbyte << 16);
+	val &= 0x7FFFFF;                  /* dismiss highest bit as it is unusable */
```

Comment:
More readable version, a compiler would know, what conversion should be applied here:
```suggestion
	uint32_t val;

	val = lbyte | (mbyte << 8) | (hbyte << 16);
	/* dismiss highest bit as it is unusable */
	val &= 0x7FFFFF;
```